src/server/pkg/backoff/retry_test.go

```diff
@@ -71,6 +71,39 @@ func TestRetryUntilCancelZeroBackoff(t *testing.T) {
 	})
 }
 
+// if mustResetBackOff is called more than once without Reset() being called in
+// between, then it returns Stop (i.e. it can back off exactly once). This is
+// used by TestRetryUntilCancelResetsOnErrContinue
+type mustResetBackOff int
+
+func (m *mustResetBackOff) NextBackOff() time.Duration {
+	if (*m) > 0 {
+		return Stop
+	}
+	(*m)++
+	return 0
+}
+
+// Reset to initial state.
+func (m *mustResetBackOff) Reset() {
+	(*m) = 0
+}
+
+// TestRetryUntilCancelResetsOnErrContinue tests that the backoff passed to
+// RetryUntilCancel is reset if 'operation' returns ErrContinue
+func TestRetryUntilCancelResetsOnErrContinue(t *testing.T) {
+	var b mustResetBackOff = 0
+	var results []int
+	RetryUntilCancel(context.Background(), func() error {
+		results = append(results, len(results)+1)
+		if len(results) < 3 {
+			return ErrContinue
+		}
+		return nil
+	}, &b, NotifyContinue(t.Name()))
+	require.Equal(t, []int{1, 2, 3}, results)
```

Comment:
I feel like I've seen this pattern a few places in our tests, is the motivation to get more informative output compared to just incrementing a counter?

Reply:
TBH, I used the pattern without thinking too much about it. Using a counter does seem equivalent, though I feel like there's probably a better pattern than either, for checking that the iterations behaved as expected rather than just checking that they happened. If I think of something better I'll send a followup PR?